src/Hazelcast.Net/Clustering/ClusterConnections.cs

```diff
@@ -491,43 +510,40 @@ private async Task<Attempt<MemberConnection>> ConnectFirstAsync(NetworkAddress a
         }
 
         /// <summary>
-        /// Gets or opens a connection to an address.
+        /// Ensures that a connection exists to a member.
         /// </summary>
-        /// <param name="address">The address.</param>
+        /// <param name="member">The member.</param>
         /// <param name="cancellationToken">A cancellation token.</param>
-        /// <returns>The connection, if successful.</returns>
         /// <remarks>
         /// <para>This method does not throw.</para>
         /// </remarks>
-        private async Task<Attempt<MemberConnection>> GetOrConnectAsync(NetworkAddress address, CancellationToken cancellationToken)
+        private async Task<Attempt<MemberConnection>> EnsureConnectionAsync(MemberInfo member, CancellationToken cancellationToken)
         {
             // if we already have a client for that address, return the client
             // if it is active, or fail if it is not - cannot open yet another
-            // client to that same address.
-            if (_addressConnections.TryGetValue(address, out var clientConnection))
-                return Attempt.If(clientConnection.Active, clientConnection);
+            // client to that same address, we'll have to wait for the inactive
+            // connection to be removed.
+            if (_connections.TryGetValue(member.Id, out var connection))
+            {
+                var active = connection.Active;
+                HConsole.WriteLine(this, $"Found {(active?"":"non-")}active connection for member {member.Id} (at {connection.Address})");
```

Comment:
```suggestion
                HConsole.WriteLine(this, $"Found {(active ? "" : "non-")}active connection for member {member.Id} (at {connection.Address})");
```
Just to make it readable ;)